### DISCUSSION TITLE
native USB - OBJEX Link v1.7-C3

### DISCUSSION THREAD

**Discussion by Aightech** [salvatoreraccardi/OBJEX_LINK#11] [General]
Hi there, 
Thank you very much for sharing your great work on github!

I looked at your OBJEX Link v1.6-C3 RISC-V schematics and I realised you were using a cp2102 for the USB communication. Is there a particular reason you did not use the native USB of the esp32-C3 ?

Thanks again!

**Comment by salvatoreraccardi** [comment]
@Aightech Hi, thank you for your question. I have already designed v1.7-C3 without the CP2102N and some improvements. Next week I will get the PCBs delivered👍

**Comment by Aightech** [comment]
Wonderful, can't wait to see it then! 😁

**Comment by salvatoreraccardi** [comment]
https://github.com/salvatoreraccardi/OBJEX_LINK/tree/main/boards/OBJEX_LINK-Rev1.7-C3_RISC-V

**Comment by dgiampetruzzi** [comment]
Ciao Salvatore,
some time ago you wrote that the ESP32C3, via native USB, works correctly on Linux / OSX but not on Windows 10. Did you find the solution to the problem?
Thank you!

**Reply by salvatoreraccardi** [reply]
yep, here is the solution: https://docs.objex.link/other/troubleshooting/riscv-boards
